### PR TITLE
Add Bee API

### PR DESCRIPTION
**Pull request by SageSphinx63920**
This pr adds events related to Bees

### CODE REVIEW

**Review by YouHaveTrouble — COMMENTED** [review]
~~You don't need to duplicate javadoc contents. If only the return value needs description, put the description on the return value.~~

**Comment by SageSphinx63920** [comment]
So where should I remove the javadoc?

**Comment by YouHaveTrouble** [comment]
```diff
-     * Returns the location of the flower
     *
+     * @return The location of the flower the bee wants to pollinate.
```

Where there's duplicate information. You describe what you return 2 times here. Similiar in other javadocs

File: patches/server/0292-Add-Bee-Api.patch [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                     return true;
+                 } else {
+                     Bee.this.remainingCooldownBeforeLocatingNewFlower = Mth.nextInt(Bee.this.random, 20, 60);
+@@ -862,6 +863,13 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
+             this.pollinating = false;
+             Bee.this.navigation.stop();
+             Bee.this.remainingCooldownBeforeLocatingNewFlower = 200;
++
++            //Purpur start
++            if(Bee.this.savedFlowerPos != null){
```
Comment (katekerllenevich) [inline patches/server/0292-Add-Bee-Api.patch]: use tenary operator for one line

File: patches/api/0050-Add-Bee-Api.patch [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
++import org.bukkit.entity.Bee;
++import org.bukkit.entity.Entity;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when a bee starts pollinating
++ */
++public class BeeStartedPollinatingEvent extends EntityEvent {
```
Comment (katekerllenevich) [inline patches/api/0050-Add-Bee-Api.patch]: cancellable

File: patches/api/0050-Add-Bee-Api.patch [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
++import org.bukkit.entity.Bee;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a bee stops pollinating
++ */
++public class BeeStopPollinatingEvent extends EntityEvent {
```
Comment (katekerllenevich) [inline patches/api/0050-Add-Bee-Api.patch]: cancellable

File: patches/api/0050-Add-Bee-Api.patch [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
++import org.bukkit.entity.Bee;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a Bee targets a flower
++ */
++public class BeeFoundFlowerEvent extends EntityEvent {
```
Comment (katekerllenevich) [inline patches/api/0050-Add-Bee-Api.patch]: cancellable

**Comment by SageSphinx63920** [comment]
We already talked about canceling the events could lead to an endless loop with pathfinding. If you cancel the found event, pathfinding could find the same flower again. Same with the Start event. The stop event should be cancellable. I think I’m going to test, how pathfinding reacts.

**Pull request merged**